src/App.tsx

```diff
@@ -1,27 +1,67 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [buttonPressed, setButtonPressed] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    let func;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    if (buttonPressed === '1') {
+      func = getAll();
+    }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    if (buttonPressed === '2') {
+      func = get5First();
+    }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    if (buttonPressed === '3') {
+      func = getRedGoods();
+    }
+
+    if (func) {
+      func.then(data => {
+        return setGoods(data);
+      });
+    }
+  }, [buttonPressed]);
+
+  const onClick1 = () => {
+    return setButtonPressed('1');
+  };
+
+  const onClick2 = () => {
+    return setButtonPressed('2');
+  };
+
+  const onClick3 = () => {
+    return setButtonPressed('3');
+  };
```

Comment:
why do you need such a lot function.
if you will hanve thousand buttons - will you make function for each button.

suggest to make one function factory which will accept function from api and return handler funcion.
```suggestion
  const makeHandleClick = (callback) => () => callback().then(setGoods);
```